src/mongocxx/pool.cpp

```diff
@@ -39,20 +39,20 @@ void pool::_release(client* client) {
 
 pool::~pool() = default;
 
-pool::pool(const uri& uri, stdx::optional<options::ssl> ssl_options)
+pool::pool(const uri& uri, const options::client& options)
```

Comment:
I don't think we need to require `options::client`.  So if we go this way, we also need a constructor overload for just a URI.

Regarding `options::pool`, I have mixed feelings.  In Jesse's words: "defy augury".  Let's not add it until we actually know we need it.